recipes/_helper/__init__.py

```diff
@@ -1,6 +1,14 @@
-from sqlalchemy import create_engine
+from sqlalchemy import create_engine, text
+from sqlalchemy.engine import Engine
 import os
 from datetime import date
 
-EDM_DATA = create_engine(os.environ["EDM_DATA"])
 DATE = date.today().strftime("%Y-%m-%d")
+
+def create_edm_date_engine() -> Engine:
```

Comment:
What's the rationale for creating an engine whenever we want to connect to the db?

Reply:
I've also seen stuff fail when a persistent engine is used in a python script and the connection timesout

Reply:
Hmm. Not entirely sure why this would fix errors you're seeing (not saying it didn't fix things, it's just a bit odd) - the engine from my understanding really is more of a config - it doesn't actually connect to the database itself upon creation, but it's an object that manages connection pools. [Sqlalchemy docs](https://docs.sqlalchemy.org/en/20/core/connections.html#basic-usage) best practice is to just declare once per module/app/etc and re-use. 

I think if we see issues with connections, we should try to manage connections more explicitly rather than just create new engines each time. Like when errors were popping up, or timing out, were there connections not being disposed? Or not being used in a `with ... as` statement? That would sort of make sense as to why a new engine helps - just a new connection pool without any lingering issues each time

Reply:
per chat, changed in [770d72b](https://github.com/NYCPlanning/ceqr-app-data/pull/154/commits/770d72bc7a7aa38c1f9612e33cada1eb85c330cb)